contracts/collect/AaveFeeCollectModule.sol

```diff
@@ -0,0 +1,258 @@
+     *      uint256 amount: The currency total amount to levy.
+     *      address currency: The currency address, must be internally whitelisted.
+     *      address recipient: The custom recipient address to direct earnings to.
+     *      uint16 referralFee: The referral fee to set.
+     *      bool followerOnly: Whether only followers should be able to collect.
+     *      uint40 endTimestamp: The end timestamp after which collecting is impossible.
+     *
+     * @return An abi encoded bytes parameter, which is the same as the passed data parameter.
+     */
+    function initializePublicationCollectModule(
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) external override onlyHub returns (bytes memory) {
+        (
+            uint256 collectLimit,
+            uint256 amount,
+            address currency,
+            address recipient,
+            uint16 referralFee,
+            bool followerOnly,
+            uint40 endTimestamp
+        ) = abi.decode(data, (uint256, uint256, address, address, uint16, bool, uint40));
+        if (
+            !_currencyWhitelisted(currency) ||
+            recipient == address(0) ||
+            referralFee > BPS_MAX ||
+            (endTimestamp < block.timestamp && endTimestamp > 0)
+        ) revert Errors.InitParamsInvalid();
+
+        _dataByPublicationByProfile[profileId][pubId].collectLimit = collectLimit;
+        _dataByPublicationByProfile[profileId][pubId].amount = amount;
+        _dataByPublicationByProfile[profileId][pubId].currency = currency;
+        _dataByPublicationByProfile[profileId][pubId].recipient = recipient;
+        _dataByPublicationByProfile[profileId][pubId].referralFee = referralFee;
+        _dataByPublicationByProfile[profileId][pubId].followerOnly = followerOnly;
+        _dataByPublicationByProfile[profileId][pubId].endTimestamp = endTimestamp;
+
+        return data;
+    }
+
+    /**
+     * @dev Processes a collect by:
+     *  1. Ensuring the collector is a follower if followerOnly mode == true
+     *  2. Ensuring the current timestamp is less than or equal to the collect end timestamp
+     *  2. Ensuring the collect does not pass the collect limit
+     *  3. Charging a fee
+     */
+    function processCollect(
+        uint256 referrerProfileId,
+        address collector,
+        uint256 profileId,
+        uint256 pubId,
+        bytes calldata data
+    ) external override onlyHub {
+        if (_dataByPublicationByProfile[profileId][pubId].followerOnly)
+            _checkFollowValidity(profileId, collector);
+
+        uint256 endTimestamp = _dataByPublicationByProfile[profileId][pubId].endTimestamp;
+        uint256 collectLimit = _dataByPublicationByProfile[profileId][pubId].collectLimit;
+        uint256 currentCollects = _dataByPublicationByProfile[profileId][pubId].currentCollects;
+
+        if (currentCollects >= collectLimit && collectLimit != 0) {
```

Comment:
If you want to save 3 of gas :coin: :oil_drum:  
```suggestion
        if (currentCollects == collectLimit && collectLimit != 0) {
```

Also, if we think it's more likely people using unlimited than limited, might worth to invert it (same with the timestamp I guess):

```suggestion
        if (collectLimit != 0 && currentCollects == collectLimit) {
```